pinot-spi/src/main/java/org/apache/pinot/spi/stream/MessageBatch.java

```diff
@@ -83,9 +83,7 @@ public LegacyStreamMessage(byte[] value, StreamMessageMetadata metadata) {
    * Returns the metadata associated with the message at a particular index. This typically includes the timestamp
    * when the message was ingested by the upstream stream-provider and other relevant metadata.
    */
-  default RowMetadata getMetadataAtIndex(int index) {
```

Comment:
Removing this might break existing implementations of this `MessageBatch` spi?

Reply:
I looked everywhere. There was one FakeStreamMessageBatch (which I took care of).  Frankly, it would have been better if the default wasn't there because then I would have the error and the implementation of this method is pretty trivial.